Plugins/Public/base_plugin/Main.cpp

```diff
@@ -2875,6 +2927,46 @@ void Plugin_Communication_CallBack(PLUGIN_MESSAGE msg, void* data)
 			base->Save();
 		}
 	}
+	else if (msg == CUSTOM_BASE_LAST_DOCKED)
+	{
+		LAST_PLAYER_BASE_NAME_STRUCT* info = reinterpret_cast<LAST_PLAYER_BASE_NAME_STRUCT*>(data);
+		if (clients.count(info->clientID))
+		{
+			uint lastBaseID = clients[info->clientID].last_player_base;
+			if (player_bases.count(lastBaseID))
+			{
+				info->lastBaseName = player_bases[lastBaseID]->basename;
+			}
+			else
+			{
+				info->lastBaseName = L"Destroyed Player Base";
+			}
+		}
+		else
+		{
+			info->lastBaseName = L"Object Unknown";
+		}
+	}
+	else if (msg == CUSTOM_BASE_GET_NAME)
```

Comment:
Something is fishy here - why is the code in here a duplicate of the above? I would say put it in an OR and just do it once, but I doubt it makes sense to do that